Function for similar words/corrections

scymtym opened the pull request.

phoe commented on test/english.lisp:
```diff
@@ -84,6 +84,18 @@
     (one-variant "Circumstanzes")
     (one-variant "CIRCUMSTANZES")))
 
+(spell:english-corrections "circumstances" :threshold 0)
+;;; => ("circumstances") T
+(spell:english-corrections "circumstances" :threshold 1)
+;;; => ("circumstances" "circumstance" "circumstances'" "circumstance's") T
+
+(spell:english-corrections "circumstanzes" :threshold 0)
+;;; => () NIL
+(spell:english-corrections "circumstanzes" :threshold 1)
+;;; => ("circumstances") NIL
+(spell:english-corrections "circumstanzes" :threshold 2)
+;;; => ("circumstances" "circumstance" "circumstances'" "circumstance's") NIL
+
```
Mind turning these into an actual test form?

scymtym committed:
New generic functions LEAFP and INTERIORP

Using predicates that are generic functions is a bit clearer
than (typep ... 'something-mixin), more flexible and often performs
better.

scymtym committed:
Add low-level similarity traversal function NODE-MAP-SIMILAR

scymtym committed:
Add MAP-SIMILAR (and %MAP-SIMILAR) functions for DICTIONARY

scymtym committed:
Add convenience functions [MAP-]CORRECTIONS

scymtym committed:
documentation: Add MAP-SIMILAR, MAP-CORRECTIONS and CORRECTIONS

scymtym committed:
Extract function MAP-ENGLISH-CASE-VARIANTS

In addition to ENGLISH-LOOKUP, correction related functions will use
this.

scymtym committed:
Add convenience function ENGLISH-CORRECTIONS

scymtym committed:
documentation: Add ENGLISH-CORRECTIONS

scymtym committed:
data: Add entry for similarity and corrections functions in changes.sexp

scymtym committed:
Update README.org
